integration-tests/bats/sql-merge.bats

```diff
@@ -1130,6 +1130,173 @@ SQL
     [[ "$output" =~ "$regex" ]] || false
 }
 
+@test "sql-merge: ff merge doesn't stomp working changes" {
+  dolt sql <<SQL
+CREATE TABLE test2 (pk int);
+CALL DOLT_COMMIT('-m', 'initial commit');
+CALL DOLT_CHECKOUT('-b', 'merge_branch');
+INSERT INTO test VALUES (9);
+CALL DOLT_COMMIT('-a', '-m', 'modify test');
+CALL DOLT_CHECKOUT('main');
+INSERT INTO test2 VALUES (8);
+SQL
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "test2" ]] || false
+    [[ ! "$output" =~ "test1" ]] || false
+
+    run dolt sql -q "CALL DOLT_MERGE('merge_branch');"
+    log_status_eq 0
+
+    run dolt sql -q "select COUNT(*) from test"
+    [[ "$output" =~ 4 ]] || false
+
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "test2" ]] || false
+    [[ ! "$output" =~ "test1" ]] || false
+}
+
+@test "sql-merge: no-ff merge doesn't stomp working changes and doesn't fast forward" {
+    dolt commit -m "initial commit"
+    dolt checkout -b merge_branch
+    dolt SQL -q "INSERT INTO test values (9)"
+    dolt add test
+    dolt commit -m "modify test"
+
+    dolt checkout main
+    dolt SQL -q "CREATE TABLE t2 (pk int)"
+    dolt SQL -q "INSERT INTO t2 values (0)"
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test" ]] || false
+
+    run dolt sql -q "CALL DOLT_MERGE('merge_branch', '--no-ff', '-m', 'no-ff merge');"
+    log_status_eq 0
+    [[ ! "$output" =~ "| 1            |" ]] || false
+
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test" ]] || false
+
+    run dolt log
+    log_status_eq 0
+    [[ "$output" =~ "no-ff merge" ]] || false
+}
+
+@test "sql-merge: squash merge" {
+    dolt commit -m "initial commit"
+    dolt checkout -b merge_branch
+    dolt SQL -q "INSERT INTO test values (3)"
+    dolt add test
+    dolt commit -m "add pk 3 to test"
+
+    dolt checkout main
+    dolt SQL -q "INSERT INTO test values (4)"
+    dolt add test
+    dolt commit -m "add pk 4 to test1"
+
+    dolt SQL -q "CREATE TABLE t2 (pk int)"
+    dolt SQL -q "INSERT INTO t2 values (9)"
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test1" ]] || false
+
+    run dolt sql -q "CALL DOLT_MERGE('--squash', 'merge_branch', '--no-commit');"
+    log_status_eq 0
+    [[ "$output" =~ "| 0            | 0         |" ]] || false
```

Comment:
Adding the header line would make the test more readable as a table. Some tests verify every line of output is there, but I'm not a huge fan of that.